src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The `useSelector` hook does not have a `withTypes` method. Instead, you should create a typed version of `useSelector` like this: `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;`